### PR TITLE
Fix C# warnings across the project, modernize slightly

### PR DESCRIPTION
**Pull request by AtariDreams**
No behavioral changes, just C# modernizations.

### CODE REVIEW

**Review by lhecker — APPROVED** [review]

**Comment by AtariDreams** [comment]
Ready!

**Comment by AtariDreams** [comment]
@lhecker Ready and everything is fixed + resolved!

**Comment by AtariDreams** [comment]
Any updates on this?

**Comment by lhecker** [comment]
@AtariDreams If you don't rebase again, it'll be reviewed and merged soon.

File: src/cascadia/WindowsTerminal_UIATests/Elements/TerminalApp.cs [hunk 1]
```diff
@@ -166,11 +166,11 @@ private void ExitProcess()
...
             this.UIRoot = null;
 
-            if (this.job != IntPtr.Zero)
+            if (this.job == IntPtr.Zero)
             {
-                WinBase.TerminateJobObject(this.job, 0);
+                return;
             }
-            this.job = IntPtr.Zero;
+            WinBase.TerminateJobObject(this.job, 0);
```
Comment (lhecker) [inline src/cascadia/WindowsTerminal_UIATests/Elements/TerminalApp.cs:173]: This isn't equivalent due to the lack of `this.job = ...`. I'd suggest:
```suggestion
            if (this.job != IntPtr.Zero)
            {
                WinBase.TerminateJobObject(this.job, 0);
                this.job = IntPtr.Zero;
            }
```

File: src/cascadia/WpfTerminalControl/TerminalContainer.cs [hunk 2]
```diff
@@ -275,10 +277,11 @@ internal void RaiseResizedIfDrawSpaceIncreased()
         /// <inheritdoc/>
         protected override void OnDpiChanged(DpiScale oldDpi, DpiScale newDpi)
         {
-            if (this.terminal != IntPtr.Zero)
+            if (this.terminal == IntPtr.Zero)
             {
-                NativeMethods.TerminalDpiChanged(this.terminal, (int)(NativeMethods.USER_DEFAULT_SCREEN_DPI * newDpi.DpiScaleX));
+                return;
             }
+            NativeMethods.TerminalDpiChanged(this.terminal, (int)(NativeMethods.USER_DEFAULT_SCREEN_DPI * newDpi.DpiScaleX));
```
Comment (lhecker) [inline src/cascadia/WpfTerminalControl/TerminalContainer.cs:284]: I would prefer the previous version here. The early return doesn't help in this case.

File: src/host/ft_uia/Elements/CmdApp.cs [hunk 3]
```diff
@@ -410,10 +410,11 @@ private void ExitCmdProcess()
...
             this.UIRoot = null;
 
-            if (this.job != IntPtr.Zero)
+            if (this.job == IntPtr.Zero)
             {
-                WinBase.TerminateJobObject(this.job, 0);
+                return;
             }
+            WinBase.TerminateJobObject(this.job, 0);
             this.job = IntPtr.Zero;
```
Comment (lhecker) [inline src/host/ft_uia/Elements/CmdApp.cs:418]: Same here. I feel like this is easier to read:
```suggestion
            if (this.job != IntPtr.Zero)
            {
                WinBase.TerminateJobObject(this.job, 0);
                this.job = IntPtr.Zero;
            }
```

**Comment by AtariDreams** [comment]
Done @lhecker

**Review by lhecker — APPROVED** [review]

**Review by carlos-zamora — APPROVED** [review]
Thanks for doing this!

**Comment by carlos-zamora** [comment]
@msftbot merge this in 10 minutes

File: src/cascadia/WpfTerminalControl/TerminalTheme.cs [hunk 4]
```diff
@@ -58,32 +58,32 @@ public struct TerminalTheme
         /// <summary>
         /// The default background color of the terminal, represented in Win32 COLORREF format.
         /// </summary>
-        public uint DefaultBackground;
+        public readonly uint DefaultBackground;
```
Comment (DHowett) [inline src/cascadia/WpfTerminalControl/TerminalTheme.cs:61]: This changes the public API contract of `TerminalTheme`, doesn't it? We don't know how our consumers are using it.

**Review by DHowett — CHANGES_REQUESTED** [review]
Couple questions, one about code flow and the other about public API contracts.

File: src/cascadia/WpfTerminalControl/TerminalContainer.cs [hunk 5]
```diff
@@ -40,18 +40,20 @@ public TerminalContainer()
...
-                this.blinkTimer = new DispatcherTimer();
-                this.blinkTimer.Interval = TimeSpan.FromMilliseconds(blinkTime);
-                this.blinkTimer.Tick += (_, __) =>
-                {
-                    if (this.terminal != IntPtr.Zero)
-                    {
-                        NativeMethods.TerminalBlinkCursor(this.terminal);
-                    }
-                };
+                return;
```
[the remark this answers is not in the document]
Comment (AtariDreams) [reply src/cascadia/WpfTerminalControl/TerminalContainer.cs:45]: I just thought it looked better to have less nesting.

File: src/cascadia/WpfTerminalControl/TerminalTheme.cs (hunk 4 above)
Comment (AtariDreams) [reply src/cascadia/WpfTerminalControl/TerminalTheme.cs:61]: Addressed!

**Comment by AtariDreams** [comment]
@DHowett Done

File: src/cascadia/WpfTerminalControl/TerminalTheme.cs (hunk 4 above)
Comment (DHowett) [reply src/cascadia/WpfTerminalControl/TerminalTheme.cs:61]: I'm sorry, I'm not totally certain about the fix here. I saw one commit that removed `readonly` from a DispatcherTimer, but in the final PR the fields in `TerminalTheme` (which is part of our public API) are still marked `readonly`.

**Review by DHowett — CHANGES_REQUESTED** [review]

**Review by DHowett — APPROVED** [review]
Thanks!

**Pull request merged**